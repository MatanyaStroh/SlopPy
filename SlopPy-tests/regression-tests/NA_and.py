''' 
  (NA and False) = False 
  (False and NA) = False

  Sources: http://databases.about.com/cs/sql/a/aa042803a.htm
           http://www.sqlite.org/nulls.html
''' 

x = 1 / 0
assert type(x) is NA
assert (x and False) is False
assert (False and x) is False

assert (x and False and x) is False
assert (x and x and False) is False
assert (False and x and x) is False

