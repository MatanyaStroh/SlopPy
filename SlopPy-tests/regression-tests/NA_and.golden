{NA <type 'exceptions.ZeroDivisionError'> 'integer division or modulo by zero'
 Traceback (most recent call last):
  File "regression-tests/NA_and.py", line 9, in <module>
    x = 1 / 0

Locals: {'__builtins__': <module '__builtin__' (built-in)>, '__name__': '__main__', '__file__': 'regression-tests/NA_and.py', '__doc__': ' \n  (NA and False) = False \n  (False and NA) = False\n\n  Sources: http://databases.about.com/cs/sql/a/aa042803a.htm\n           http://www.sqlite.org/nulls.html\n', '__package__': None}
}
{JUMP_IF_FALSE(NA) no jump | regression-tests/NA_and.py <module>:11}
{JUMP_IF_FALSE(NA) no jump | regression-tests/NA_and.py <module>:14}
{JUMP_IF_FALSE(NA) no jump | regression-tests/NA_and.py <module>:15}
{JUMP_IF_FALSE(NA) no jump | regression-tests/NA_and.py <module>:15}
